feat: /get_answer endpoint 만들기

2021110806 opened the pull request:
DB_PATH 경로를 제 로컬에 맞게 수정해서, 실행 시 이 부분 확인 부탁드립니다.

pdf 파일 관련 UI, 함수들 주석 처리 한 것들이 있습니다. 이 부분 나중에 최종 버전에서는 해제해야 할 것으로 보입니다.

vite.config.js 포트 번호 수정 했는데 필요했던 부분인지 모르겠습니다.

2021110806 committed:
feat: /get_answer endpoint 만들기

sprout88 commented:
vite.config.js 에서 포트번호를 3000 에서 5173 으로 변경하신듯 합니다. 5173이 명시하지 않으면 설정되는 vite 기본 개발서버 포트인데, 혹시 어떤 이유때문에 변경하신지 궁금합니다. 따로 이유가 없으면 다른 분들이 햇갈리실 수 있으니 다시 3000 포트를 사용하도록 되돌리는게 좋을 것 같습니다

sprout88 commented:
![image](https://github.com/user-attachments/assets/2ec2d1ed-e462-432a-910b-90f9320d1c85)

이 DB_PATH 부분을 확인해보았는데,
제 로컬환경과 어떤 부분이 달라서 오류가 생기는지는 모르겠으나,
이 수정본은 제 로컬환경에서 오류를 일으킵니다.

첨부한 사진의 DEBUG 탭을 보시면, /backend/backend/vectorDB 를 참조하려고 해서 오류가 생긴 것을 볼 수 있습니다.

우선, {script_dir}/../../backend/vectorDB 와 같은 절대경로와 상대경로가 합쳐진 이상한 경로 표현법은 문제가 있어보입니다. 

이 부분이 각자의 개발환경에서 올바르게 동작할 수 있도록 수정해야할 것 같습니다.

sprout88 commented on backend/src/app.py:
```diff
@@ -1,13 +1,25 @@
 
 app = FastAPI(debug=True)
 
+# CORS 미들웨어 추가
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=["http://localhost:5173"],  # 프론트엔드 주소
+    allow_credentials=True,
+    allow_methods=["*"],  # 모든 HTTP 메서드 허용
+    allow_headers=["*"],  # 모든 헤더 허용
+)
+
```
이 CORSMiddleware 는 불필요해보이는데, 어떤 이유로 추가하신 것인지 궁금합니다.

sprout88 commented on frontend/my-svelte-app/dist/build/main.js:
어.. 그리고 빌드 디렉토리에 있는 코드가 커밋에 포함된 것 같습니다. 의도된 커밋인지 궁금합니다

sprout88 committed:
fix: 경로 문제 수정

test_main.py 가 위치한 디렉토리를 변경해서 경로 참조 불가 문제를 해결했습니다.

변경 전 : /backend/src/rag_utils/test_main.py
변경 후 : /backend/src/test_main.py

그리고 모듈들을 참조할 때, 모듈의 폴더명을 붙여야 합니다.
예시)

from rag_utils.retrieval import Retriev_Gen

sprout88 committed:
refactor: rag 객체에 db_path 추가 + 테스트 데이터 위치 변경

rag_utils 의 객체들이 test_main.py 테스트 코드로부터 db_path 를 전달받도록 클래스를 수정했습니다.

rag_utils 에 rag 와 관련된 모듈(python 코드)만 모아놓기 위해서 test_data 폴더에 테스트데이터를 놓도록 수정했습니다.

sprout88 committed:
chore: gitignore 에 vectorDB 디렉토리 추가

sprout88 committed:
chore: gitignore 에 vectorDB 디렉토리 추가 - 수정

.gitignore 에서 vectorDB 를 무시하는 설정의 위치를 변경했습니다. (이후 프로젝트 구조가 변경될 것에 대비)

sprout88 committed:
chore: 배포용 빌드를 제거

sprout88 committed:
chore: 프론트엔드 개발서버의 포트번호를 명시

sprout88 committed:
fix: 잘못된 api 요청 경로 수정

sprout88 committed:
refactor: rag_utils 의 바뀐 경로 참조 적용

sprout88 committed:
fix: 불필요한 CORS 미들웨어 제거

sprout88 reviewed (approved):
작동확인되어서 머지 승인합니다
